src/common/tse_internal.h

```diff
@@ -70,7 +70,7 @@ struct tse_task_private {
 					 dtp_completing:1,
 					/* task is in running state */
 					 dtp_running:1,
-					 dtp_dep_cnt:30;
+					 dtp_dep_cnt:29;
```

Comment:
What is the reason for this?

Reply:
It is a bug in current master, I suspected that it is the root reason for DAOS-2846, so fix it. But it does not resolve DAOS-2846.

Reply:
It's definitely a bad idea to make the "dep_cnt" sharing same uint32_t with  'flags', I think we need to fix this. (use separate uint32_t for 'dep_cnt' and 'flags')

We also need to make sure current code holding lock probably when changing the 'count' & 'flags'.